chia/server/address_manager.py

```diff
@@ -443,8 +443,9 @@ def select_peer_(self, new_only: bool) -> Optional[ExtendedPeerInfo]:
         if not new_only and self.tried_count > 0 and (self.new_count == 0 or randrange(2) == 0):
             chance = 1.0
             start = time.time()
+            cached_tried_matrix_positions: List[Tuple[int, int]] = []
             if len(self.used_tried_matrix_positions) < math.sqrt(TRIED_BUCKET_COUNT * BUCKET_SIZE):
-                cached_tried_matrix_positions = list(self.used_tried_matrix_positions)
+                cached_tried_matrix_positions += self.used_tried_matrix_positions
```

Comment:
as someone who isn't a python expert, this is not obviously correct. If `used_tried_matrix_position` is iterable, the `list()` constructor will convert it to a list, whereas `+=` will just append it. Actually, the exact semantics of the `+=` operator when the right hand side is an iterable is not trivial to find documentation for.

As far as I can tell, the `+=` operator on lists behaves different than `A = A + B`, which seems quite unintuitive.

I would lean towards leaving these assignment as they were, but still declare the empty list above.

Reply:
hi @arvidn, updated.